src/github.com/matrix-org/go-neb/services/github/github.go

```diff
@@ -77,14 +79,69 @@ func (s *Service) requireGithubClientFor(userID string) (cli *gogithub.Client, r
 	return
 }
 
+const numberGithubSearchSummaries = 3
+const cmdGithubSearchUsage = `!github create owner/repo "search query"`
+
+func (s *Service) cmdGithubSearch(roomID, userID string, args []string) (interface{}, error) {
+	cli := s.githubClientFor(userID, true)
+	if len(args) < 2 {
+		return &gomatrix.TextMessage{"m.notice", "Usage: " + cmdGithubSearchUsage}, nil
+	}
+
+	query := fmt.Sprintf("repo:%s %s", args[0], strings.Join(args[1:], " "))
+	searchResult, res, err := cli.Search.Issues(query, nil)
+
+	if err != nil {
+		log.WithField("err", err).Print("Failed to search")
+		if res == nil {
+			return nil, fmt.Errorf("Failed to search. Failed to connect to Github")
+		}
+		return nil, fmt.Errorf("Failed to search. HTTP %d", res.StatusCode)
+	}
+
+	if searchResult.Total == nil || *searchResult.Total == 0 {
+		return &gomatrix.TextMessage{"m.notice", "No results found for your search query!"}, nil
+	}
+
+	numResults := *searchResult.Total
+	numberOfSummaries := numberGithubSearchSummaries
+	if numResults < numberGithubSearchSummaries {
+		numberOfSummaries = numResults
+	}
+
+	summarizedIssues := searchResult.Issues[0:numberOfSummaries]
+
+	var htmlBuffer bytes.Buffer
+	var plainBuffer bytes.Buffer
+	htmlBuffer.WriteString(fmt.Sprintf("Found %d results, here are the most relevant:<br><ol>", numResults))
+	plainBuffer.WriteString(fmt.Sprintf("Found %d results, here are the most relevant:\n", numResults))
+	for i, issue := range summarizedIssues {
+		if issue.HTMLURL == nil || issue.User.Login == nil || issue.Title == nil {
+			continue
```

Comment:
It's unfortunate that we apply the max issue check and then have this additional check which may result in even fewer entries. I'd be tempted to just loop all the `searchResult.Issues` and stop once you've added `numberGithubSearchSummaries` or reached the end of the slice.

Reply:
done